lib/config/config-file.js

```diff
@@ -360,8 +360,9 @@ function resolve(filePath) {
             var scopedPackageShortcutRegex = /^(@[^\/]+)(?:\/(?:eslint-config)?)?$/;
             if (scopedPackageShortcutRegex.test(filePath)) {
                 filePath = filePath.replace(scopedPackageShortcutRegex, "$1/eslint-config");
-            } else if (filePath.split("/")[1].indexOf("eslint-config-") !== 0) {
-                // for scoped packages, insert the eslint-config after the first /
+            } else if (!/^eslint-config(-|$)/.test(filePath.split("/")[1])) {
```

Comment:
@ilyavolodin `@something/eslint-config` would be the package, `foo.js` would be a file inside it.